.changesets/fix_geal_cache_key_qp_options.md

```diff
@@ -0,0 +1,32 @@
+### Prevent query plan cache collision when planning options change ([Issue #5093](https://github.com/apollographql/router/issues/5093))
+
+When query planning takes place there are a number of options such as:
+* `defer_support`
+* `generate_query_fragments`
+* `experimental_reuse_query_fragments`
+* `experimental_type_conditioned_fetching`
+* `experimental_query_planner_mode`
+
+that will affect the generated query plans.
+
+If distributed query plan caching is also enabled, then changing any of these will result in different query plans being generated and entering the cache.
+
+This could cause issue in the following scenarios:
+1. The Router configuration changes and a query plan is loaded from cache which is incompatible with the new configuration.
+2. Routers with differing configuration are sharing the same cache causing them to cache and load incompatible query plans. 
+
+Now a hash for the entire query planner configuration is included in the cache key to prevent this from happening.
+
+If you are running previous versions of the Router and wish to change the query planner options then you must take care to
+change the cache namespace to prevent collisions.
+
```

Comment:
I'm trying to build my own clarity here — is this a changelog entry that's meant to inform previous users of the Router, or is it still somehow relevant after this fix?   My understanding is that this fix makes that no longer a concern, but previous users would have had to leverage a custom cache key prefix.  Is that right?

If my understanding is correct, I would just remove line 20 through the of the example here because it makes it sound like _using_ this cache key prefix is necessary if I was coming from a previous version of the Router.

Reply:
Yes, it's flagging to users of the old versions that they must change the namespace

Reply:
[4968a88](https://github.com/apollographql/router/pull/5100/commits/4968a88a5d5d4d351b9df67b1c241bdcfaf07f94)